### PR TITLE
ENG-9494 expose roles in request context

### PR DESCRIPTION
**Pull request by d-trace**
## Description
Add roles parsing logic to JWT parser and expose roles in request context.
The main motivation is to be able to reuse this logic in other parts of codebase, specifically in graphql context

### Testing
Added unit tests

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

**Review by surajpuvvada — DISMISSED** [review]
lgtm - but pls wait for one more review

File: grpc-context-utils/src/main/java/org/hypertrace/core/grpcutils/context/JwtParser.java
Comment (aaron-steinfeld) [inline grpc-context-utils/src/main/java/org/hypertrace/core/grpcutils/context/JwtParser.java:90]: Can we use a predictable iteration set (or a list)

File: grpc-context-utils/src/test/java/org/hypertrace/core/grpcutils/context/JwtParserTest.java
Comment (aaron-steinfeld) [inline grpc-context-utils/src/test/java/org/hypertrace/core/grpcutils/context/JwtParserTest.java:27]: We can drop the traceable role - want to leak as little as possible here

Comment (d-trace) [reply grpc-context-utils/src/test/java/org/hypertrace/core/grpcutils/context/JwtParserTest.java]: Sure. Done

File: grpc-context-utils/src/main/java/org/hypertrace/core/grpcutils/context/JwtParser.java
Comment (d-trace) [reply grpc-context-utils/src/main/java/org/hypertrace/core/grpcutils/context/JwtParser.java]: Done

**Comment by d-trace** [comment]
@aaron-steinfeld addressed comments. Please have another look

**Review by avinashkolluru — APPROVED** [review]

**Comment by d-trace** [comment]
hey @aaron-steinfeld, addressed issues we discussed offline. Please have a look

**Review by aaron-steinfeld — APPROVED** [review]

**Review by surajpuvvada — APPROVED** [review]

**Pull request merged**
